### ISSUE TITLE
Cannot install gtk2hs-buildtools with stack

### ISSUE THREAD

**Issue by mwu-tow** [commercialhaskell/stack#1050]
When I give the `cabal install gtk2hs-buildtools` correctly installs the package. However, the same fails with stack:
`stack install gtk2hs-buildtools` fails with an error:
```
    Linking .stack-work\dist\x86_64-windows\Cabal-1.22.4.0\build\gtk2hsTypeGen\g
tk2hsTypeGen.exe ...
    Preprocessing executable 'gtk2hsHookGenerator' for
    gtk2hs-buildtools-0.13.0.4...
    [1 of 2] Compiling Paths_gtk2hs_buildtools ( .stack-work\dist\x86_64-windows
\Cabal-1.22.4.0\build\autogen\Paths_gtk2hs_buildtools.hs, .stack-work\dist\x86_6
4-windows\Cabal-1.22.4.0\build\gtk2hsHookGenerator\gtk2hsHookGenerator-tmp\Paths
_gtk2hs_buildtools.o )
    .stack-work\dist\x86_64-windows\Cabal-1.22.4.0\build\gtk2hsHookGenerator\gtk
2hsHookGenerator-tmp\.stack-work\dist\x86_64-windows\Cabal-1.22.4.0\build\autoge
n\Paths_gtk2hs_buildtools.dump-hi: openFile: does not exist (No such file or dir
ectory)
```

I give the command from my home directory, outside from any projects.

See https://gist.github.com/mwu-tow/2c7f9f204eed5efedd16 for the full output
and https://gist.github.com/mwu-tow/c1e71813b76443ecc794 for the `--verbose` version

My stack `--version` is `Version 0.1.4.0, Git revision dac8cfe51802e08edb523b80c51cf35e21933513 (1722 commits) X86_64` — but I just checked the recently released 0.1.5.0 and the issue is the same.

Interestingly, on another Windows machine it works fine. 
Could this be caused by the Windows paths length limitation? Where is the file that cannot be opened? (only relative path is outputted, I'm not sure where I should look for it)

**Comment by snoyberg** [comment]
Can you try first `set STACK_ROOT=c:\sw` to see if it helps? In all likelihood, it won't, and this will be triggered by the length of your temporary directory path.

**Comment by snoyberg** [comment]
I just merged #1027, which should help with this. If you try again with Stack master (`stack upgrade --git`), it will likely succeed.

**Comment by mwu-tow** [comment]
I already have `STACK_ROOT=c:\s` so it does not help. 

The master version solved the problem for me. :-)
